From 8913f60347802077881c4829c3c10ee589b0870f Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Thu, 25 Oct 2018 13:02:22 +0200
Subject: [PATCH 15/17] doc: Fix dist not working

automake's documentation says that it works depth-first, but it doesn't
actually go depth first when dist'ing a directory. So our .ref.xml files
aren't generated when we try to get them added to the dist.

Fix this with a big hammer by moving the generation to the current
directory, rather than spending any more time working with automake.
Can't wait for the meson port.
---
 Makefile.am                             |  2 +-
 configure.ac                            |  1 -
 doc/Makefile.am                         | 35 ++++++++++++++++++++-----
 doc/{dbus => }/dbus-introspect-docs.dtd |  0
 doc/dbus/Makefile.am                    | 27 -------------------
 doc/{dbus => }/spec-to-docbook.xsl      |  0
 doc/upower-docs.xml                     |  6 ++---
 7 files changed, 33 insertions(+), 38 deletions(-)
 rename doc/{dbus => }/dbus-introspect-docs.dtd (100%)
 delete mode 100644 doc/dbus/Makefile.am
 rename doc/{dbus => }/spec-to-docbook.xsl (100%)

diff --git a/Makefile.am b/Makefile.am
index 13dc902..559c4f9 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -52,7 +52,7 @@ pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = upower-glib.pc
 
 # xsltproc barfs on 'make distcheck'; disable for now
-DISTCHECK_CONFIGURE_FLAGS=--disable-man-pages --enable-gtk-doc			\
+DISTCHECK_CONFIGURE_FLAGS=--enable-man-pages --enable-gtk-doc			\
 	--with-systemdsystemunitdir=$$dc_install_base/$(systemdsystemunitdir)	\
 	--with-udevrulesdir='$${libdir}/udev/rules.d-distcheck'
 
diff --git a/configure.ac b/configure.ac
index 3df3055..ad8f941 100644
--- a/configure.ac
+++ b/configure.ac
@@ -253,7 +253,6 @@ tools/Makefile
 doc/Makefile
 doc/version.xml
 doc/man/Makefile
-doc/dbus/Makefile
 rules/Makefile
 libupower-glib/Makefile
 libupower-glib/up-version.h
diff --git a/doc/Makefile.am b/doc/Makefile.am
index d11a150..2c20415 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,5 +1,5 @@
 
-SUBDIRS = man dbus
+SUBDIRS = man
 
 NULL =
 
@@ -50,15 +50,40 @@ MKDB_OPTIONS=--sgml-mode --output-format=xml
 # Extra options to supply to gtkdoc-mktmpl
 MKTMPL_OPTIONS=
 
+org.freedesktop.UPower.ref.xml : $(top_srcdir)/dbus/org.freedesktop.UPower.xml $(top_srcdir)/doc/spec-to-docbook.xsl
+	if $(AM_V_P); then set -x; else echo " GEN   $@"; fi
+	echo "<?xml version=\"1.0\"?>""<!DOCTYPE refentry PUBLIC \"-//OASIS//DTD DocBook XML V4.1.2//EN\" \"http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd\">" > $@
+	$(XSLTPROC) $(top_srcdir)/doc/spec-to-docbook.xsl $< | tail -n +2 >> $@
+
+org.freedesktop.UPower.Device.ref.xml : $(top_srcdir)/dbus/org.freedesktop.UPower.Device.xml $(top_srcdir)/doc/spec-to-docbook.xsl
+	if $(AM_V_P); then set -x; else echo " GEN   $@"; fi
+	echo "<?xml version=\"1.0\"?>""<!DOCTYPE refentry PUBLIC \"-//OASIS//DTD DocBook XML V4.1.2//EN\" \"http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd\">" > $@
+	$(XSLTPROC) $(top_srcdir)/doc/spec-to-docbook.xsl $< | tail -n +2 >> $@
+
+org.freedesktop.UPower.KbdBacklight.ref.xml : $(top_srcdir)/dbus/org.freedesktop.UPower.KbdBacklight.xml $(top_srcdir)/doc/spec-to-docbook.xsl
+	if $(AM_V_P); then set -x; else echo " GEN   $@"; fi
+	echo "<?xml version=\"1.0\"?>""<!DOCTYPE refentry PUBLIC \"-//OASIS//DTD DocBook XML V4.1.2//EN\" \"http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd\">" > $@
+	$(XSLTPROC) $(top_srcdir)/doc/spec-to-docbook.xsl $< | tail -n +2 >> $@
+
+org.freedesktop.UPower.Wakeups.ref.xml : $(top_srcdir)/dbus/org.freedesktop.UPower.Wakeups.xml $(top_srcdir)/doc/spec-to-docbook.xsl
+	if $(AM_V_P); then set -x; else echo " GEN   $@"; fi
+	echo "<?xml version=\"1.0\"?>""<!DOCTYPE refentry PUBLIC \"-//OASIS//DTD DocBook XML V4.1.2//EN\" \"http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd\">" > $@
+	$(XSLTPROC) $(top_srcdir)/doc/spec-to-docbook.xsl $< | tail -n +2 >> $@
+
+EXTRA_DIST = spec-to-docbook.xsl dbus-introspect-docs.dtd
+
+BUILT_SOURCES =							\
+	org.freedesktop.UPower.ref.xml			\
+	org.freedesktop.UPower.Device.ref.xml		\
+	org.freedesktop.UPower.KbdBacklight.ref.xml
+
 # Non-autogenerated SGML files to be included in $(DOC_MAIN_SGML_FILE)
 content_files =  			    				\
 	version.xml			    				\
 	man/upower.xml		    					\
 	man/upowerd.xml		    					\
 	man/UPower.xml		    					\
-	dbus/org.freedesktop.UPower.ref.xml				\
-	dbus/org.freedesktop.UPower.Device.ref.xml			\
-	dbus/org.freedesktop.UPower.KbdBacklight.ref.xml		\
+	$(BUILT_SOURCES)						\
 	$(NULL)
 
 # Images to copy into HTML directory
@@ -84,8 +109,6 @@ DISTCLEANFILES =					\
 
 if ENABLE_GTK_DOC
 include $(top_srcdir)/gtk-doc.make
-else
-EXTRA_DIST =
 endif
 
 # Version information for marking the documentation
diff --git a/doc/dbus/dbus-introspect-docs.dtd b/doc/dbus-introspect-docs.dtd
similarity index 100%
rename from doc/dbus/dbus-introspect-docs.dtd
rename to doc/dbus-introspect-docs.dtd
diff --git a/doc/dbus/Makefile.am b/doc/dbus/Makefile.am
deleted file mode 100644
index 78c12fb..0000000
--- a/doc/dbus/Makefile.am
+++ /dev/null
@@ -1,27 +0,0 @@
-
-all : org.freedesktop.UPower.ref.xml org.freedesktop.UPower.Device.ref.xml org.freedesktop.UPower.KbdBacklight.ref.xml org.freedesktop.UPower.Wakeups.ref.xml
-
-org.freedesktop.UPower.ref.xml : $(top_srcdir)/dbus/org.freedesktop.UPower.xml $(top_srcdir)/doc/dbus/spec-to-docbook.xsl
-	echo "<?xml version=\"1.0\"?>""<!DOCTYPE refentry PUBLIC \"-//OASIS//DTD DocBook XML V4.1.2//EN\" \"http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd\">" > $@
-	$(XSLTPROC) $(top_srcdir)/doc/dbus/spec-to-docbook.xsl $< | tail -n +2 >> $@
-
-org.freedesktop.UPower.Device.ref.xml : $(top_srcdir)/dbus/org.freedesktop.UPower.Device.xml $(top_srcdir)/doc/dbus/spec-to-docbook.xsl
-	echo "<?xml version=\"1.0\"?>""<!DOCTYPE refentry PUBLIC \"-//OASIS//DTD DocBook XML V4.1.2//EN\" \"http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd\">" > $@
-	$(XSLTPROC) $(top_srcdir)/doc/dbus/spec-to-docbook.xsl $< | tail -n +2 >> $@
-
-org.freedesktop.UPower.KbdBacklight.ref.xml : $(top_srcdir)/dbus/org.freedesktop.UPower.KbdBacklight.xml $(top_srcdir)/doc/dbus/spec-to-docbook.xsl
-	echo "<?xml version=\"1.0\"?>""<!DOCTYPE refentry PUBLIC \"-//OASIS//DTD DocBook XML V4.1.2//EN\" \"http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd\">" > $@
-	$(XSLTPROC) $(top_srcdir)/doc/dbus/spec-to-docbook.xsl $< | tail -n +2 >> $@
-
-org.freedesktop.UPower.Wakeups.ref.xml : $(top_srcdir)/dbus/org.freedesktop.UPower.Wakeups.xml $(top_srcdir)/doc/dbus/spec-to-docbook.xsl
-	echo "<?xml version=\"1.0\"?>""<!DOCTYPE refentry PUBLIC \"-//OASIS//DTD DocBook XML V4.1.2//EN\" \"http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd\">" > $@
-	$(XSLTPROC) $(top_srcdir)/doc/dbus/spec-to-docbook.xsl $< | tail -n +2 >> $@
-
-EXTRA_DIST = spec-to-docbook.xsl dbus-introspect-docs.dtd
-
-MAINTAINERCLEANFILES = *.ref.xml
-
-clean-local :
-	rm -f *~ *.ref.xml
-
--include $(top_srcdir)/git.mk
diff --git a/doc/dbus/spec-to-docbook.xsl b/doc/spec-to-docbook.xsl
similarity index 100%
rename from doc/dbus/spec-to-docbook.xsl
rename to doc/spec-to-docbook.xsl
diff --git a/doc/upower-docs.xml b/doc/upower-docs.xml
index cc5f890..48bbd92 100644
--- a/doc/upower-docs.xml
+++ b/doc/upower-docs.xml
@@ -64,9 +64,9 @@
 	UPower daemon.
       </para>
     </partintro>
-    <xi:include href="dbus/org.freedesktop.UPower.ref.xml"/>
-    <xi:include href="dbus/org.freedesktop.UPower.Device.ref.xml"/>
-    <xi:include href="dbus/org.freedesktop.UPower.KbdBacklight.ref.xml"/>
+    <xi:include href="org.freedesktop.UPower.ref.xml"/>
+    <xi:include href="org.freedesktop.UPower.Device.ref.xml"/>
+    <xi:include href="org.freedesktop.UPower.KbdBacklight.ref.xml"/>
   </reference>
 
   <reference id="libupower-glib">
-- 
2.19.1

